packages/devtools_app/lib/src/screens/memory/panes/chart/controller/chart_connection.dart

```diff
@@ -35,36 +34,64 @@ class ChartConnection extends DisposableController
     isAndroidChartVisible: isAndroidChartVisible,
   );
 
-  Timer? _pollingTimer;
-  bool _connected = false;
+  bool initialized = false;
+
+  DebounceTimer? _polling;
+
+  late final bool isDeviceAndroid;
+
+  void maybeInit() async {
+    if (initialized) return;
+    initialized = true;
+
+    if (!serviceConnection.serviceManager.connectedState.value.connected) {
+      isDeviceAndroid = false;
+      return;
+    }
```

Comment:
we shouldn't need this. ChartVmConnection.maybeInit should never be called if we are not connected to an app.

Reply:
We discussed it yesterday. Added comment.

Reply:
We discussed returning early if not connected. Looking back through the commits, what we had was this:
```
if (initialized || !serviceConnection.serviceManager.connectedState.value.connected) {
  return;
}
```

Reply:
Yes, and (1) I got failure for not initialized late isDeviceAndroid and (2) i believe the state of the instance here should become `initialized and disconnected`, not `not initialized`.

How does it sound?

Reply:
Yes, but i got failure because of not initialized late isDeviceAndroid.
Replaced with assert.

Reply:
if we are getting a late initialized error on `isDeviceAndroid`, then that seems like a bug we should be tracking down. We should not be checking any values on ChartVMConnection before it has been initialized.